src/Query/OutputWalker.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query;
+
+use Doctrine\ORM\Query\Exec\SqlFinalizer;
+
+/**
+ * Interface for output walkers
+ *
+ * Output walkers, like tree walkers, can traverse the DQL AST to perform
+ * their purpose.
```

Comment:
Ok, but why are they called _output_ walkers?

Reply:
Output walker is a term that comes from https://www.doctrine-project.org/projects/doctrine-orm/en/2.17/cookbook/dql-custom-walkers.html#modify-the-output-walker-to-generate-vendor-specific-sql.

Reply:
So, it's an output walker because it outputs SQL?

Reply:
The parser builds the Abstract Syntax Tree (AST) representing the DQL statement. 

Various (more than one) `TreeWalker`s can then be started on the tree. They can traverse the AST, read information from it or modify it in various ways.

But, finally, we need an `AbstractSqlExecutor` (or, with this PR, an `SqlFinalizer` that will produce the `AbstractSqlExecutor`) to actually run this query.

The job of the "output walker" – and there can be only one – is to traverse the AST, produce suitable SQL and return the `AbstractSqlExecutor`/`SqlFinalizer` that will eventually execute the SQL query. 

Does that answer the question?

Reply:
The `\Doctrine\ORM\Tools\Pagination\RootTypeWalker`, by the way, does not really produce SQL but instead returns the colum type for the `#Id` column of the root entity. But that's a hack.